lfs/gitscanner_catfilebatch.go

```diff
@@ -100,40 +139,53 @@ func (s *catFileBatchScanner) Scan() bool {
 	return true
 }
 
-func (s *catFileBatchScanner) next() (string, *WrappedPointer, error) {
+func (s *CatFileBatchScanner) Close() error {
+	if s.closeFn == nil {
+		return nil
+	}
+	return s.closeFn()
+}
+
+func (s *CatFileBatchScanner) next() (string, string, *WrappedPointer, error) {
 	l, err := s.r.ReadBytes('\n')
 	if err != nil {
-		return "", nil, err
+		return "", "", nil, err
 	}
 
 	// Line is formatted:
 	// <sha1> <type> <size>
 	fields := bytes.Fields(l)
 	if len(fields) < 3 {
-		return "", nil, errors.Wrap(fmt.Errorf("Invalid: %q", string(l)), "git cat-file --batch")
+		return "", "", nil, errors.Wrap(fmt.Errorf("Invalid: %q", string(l)), "git cat-file --batch")
 	}
 
 	blobSha := string(fields[0])
 	size, _ := strconv.Atoi(string(fields[2]))
+	sha := sha256.New()
 	buf := make([]byte, size)
-	read, err := io.ReadFull(s.r, buf)
+	read, err := io.ReadFull(io.TeeReader(s.r, sha), buf)
```

Comment:
Using the CatFileBatchScanner in `git lfs status` will cause it to load potentially huge blobs in memory. This doesn't happen in normal git scanning because the `CatFileBatchCheckScanner` filters any blobs over `lfs.blobSizeCutoff = 1024`.

Is it possible to skip the `buf` init if the the blob size here is also over that threshold?